osidb/models.py

```diff
@@ -2919,6 +2933,83 @@ def validate(self, *args, **kwargs):
         super().clean_fields(*args, **kwargs)
 
 
+class PackageManager(ACLMixinManager, TrackingMixinManager):
+    @staticmethod
+    def create_package(flaw, package, **extra_fields):
+        """
+        Return a new Package, or update an existing Package without saving.
+        Unlike other similar manager methods in this file, new Package
+        instance is saved into database to allow relationships with PackageVer.
+        """
+        try:
+            package = Package.objects.get(flaw=flaw, package=package)
+            for attr, value in extra_fields.items():
+                setattr(package, attr, value)
+            return package
+
+        except ObjectDoesNotExist:
+            # The instance must be saved into the database, so that it's possible
+            # to connect it with PackageVer, but it must otherwise be as clean
+            # as an in-memory unsaved instance created with plain
+            # Package(flaw=flaw, package=package, **extra_fields). Calling
+            # objects.create() would involve osidb's custom logic that would
+            # interfere prematurely.
+            return Package._base_manager.create(
+                flaw=flaw, package=package, **extra_fields
+            )
+
+
+class Package(ACLMixin, BugzillaSyncMixin, TrackingMixin):
+    """
+    Model representing a package with connected versions.
+
+    The model's structure allows future extensibility with features from the CVE 5.0 schema.
+    Currently, it tracks only versions corresponding to bugzilla fixed_in field, which correspond to
+    CVE 5.0's `"status": "unaffected"`.
+    References:
+    - https://github.com/CVEProject/cve-schema/blob/master/schema/v5.0/docs/versions.md
+    - https://github.com/CVEProject/cve-schema/blob/master/schema/v5.0/CVE_JSON_5.0_schema.json
+    """
+
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    flaw = models.ForeignKey(
+        Flaw,
+        related_name="package_versions",
+        on_delete=models.CASCADE,
+    )
+
+    package = models.CharField(max_length=2058)
+
+    objects = PackageManager()
+
+    # TODO: test this
```

Comment:
One simple integration test would be nice.

Reply:
Fixed in commit [1fa690c](https://github.com/RedHatProductSecurity/osidb/pull/307/commits/1fa690cba505717ebc1353e8406395ea08d4a147)